src/data/equiv/mul_add.lean

```diff
@@ -246,6 +247,23 @@ rfl
   {M N} [monoid M] [monoid N] : function.injective (to_monoid_hom : (M ≃* N) → M →* N) :=
 λ f g h, mul_equiv.ext (monoid_hom.ext_iff.1 h)
 
+/-- The multiplicative analogue of `equiv.arrow_congr` (target monoids must be commutative). -/
+@[simps apply,
+  to_additive "The additive analogue of `equiv.arrow_congr` (target monoids must be commutative)."]
+def arrow_congr {M N P Q} [monoid M] [monoid N] [comm_monoid P] [comm_monoid Q]
+  (f : M ≃* N) (g : P ≃* Q) : (M →* P) ≃* (N →* Q) :=
+{ to_fun := λ h,
+  { to_fun := λ n, g (h (f.symm n)),
+    map_one' := by simp,
+    map_mul' := by simp, },
```

Comment:
Possibly simpler as something like
```suggestion
  g.to_monoid_hom.comp (h.comp f.symm.to_monoid_hom)
```
which avoids the need to prove `map_one'` and `map_mul'`

Reply:
Done!